libcst/metadata/base_provider.py

```diff
@@ -44,9 +44,17 @@ class BaseMetadataProvider(MetadataDependent, Generic[_ProvidedMetadataT]):
     # explanation.
     _computed: MutableMapping["CSTNode", _ProvidedMetadataT]
 
-    def __init__(self) -> None:
+    is_cache_required: bool = False
+
+    def __init__(self, cache: object = None) -> None:
         super().__init__()
         self._computed = {}
+        if self.is_cache_required and not cache:
```

Comment:
This should explicitly check against `cache is None`, otherwise we will have no way of passing a cache such as a dictionary that is empty.